File: app/src/main/java/lheureux/votodroid/activities/VoteActivity.java
package lheureux.votodroid.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.RatingBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import lheureux.votodroid.R;
import lheureux.votodroid.databinding.ActivityVoteBinding;
import lheureux.votodroid.models.Question;
import lheureux.votodroid.models.Vote;
import lheureux.votodroid.services.QuestionService;
import lheureux.votodroid.services.VoteService;


public class VoteActivity extends AppCompatActivity {
    private ActivityVoteBinding binding;
    private VoteService voteService;
    private QuestionService questionService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        voteService = VoteService.getInstance(this);
        questionService = QuestionService.getInstance(this);

        binding = ActivityVoteBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        Long questionId = getIntent().getLongExtra("questionId", 0);
        Question question = questionService.get(questionId);
        TextView tvQuestion = findViewById(R.id.tvQuestion);
        EditText voterName = findViewById(R.id.editTextTextPersonName);

        tvQuestion.setText(question.text);

        for (Question q : questionService.all()) {
            System.out.println(q.questionId);
        }

        binding.submit.setOnClickListener(view1 -> {
            try {
                RatingBar rating = findViewById(R.id.rating);
                float ratingValue = rating.getRating();
                Vote vote = new Vote(ratingValue, questionId, voterName.getText().toString());

                voteService.insert(vote);
                Intent intent= new Intent();
                setResult(RESULT_OK, intent);
                finish();

            } catch (Exception e) {
                voterName.setError(e.getMessage());
            }

        });

    }
}

File: app/src/main/java/com/theodorelheureux/votodroid_android/activities/VoteActivity.kt
package com.theodorelheureux.votodroid_android.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.theodorelheureux.votodroid_android.CreateVoteMutation
import com.theodorelheureux.votodroid_android.databinding.ActivityVoteBinding
import com.theodorelheureux.votodroid_android.graphql.GQLClient
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class VoteActivity : AppCompatActivity() {
    private var binding: ActivityVoteBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityVoteBinding.inflate(
            layoutInflater
        )
        val view: View = binding!!.root
        setContentView(view)
        binding!!.voteProgressBar.visibility = View.GONE

        val questionId = intent.getStringExtra("questionId")!!
        val questionText = intent.getStringExtra("questionText")!!

        binding!!.tvQuestion.text = questionText

        binding!!.submit.setOnClickListener {
            val value = binding!!.rating.rating.toInt()

            binding!!.submit.visibility = View.GONE
            binding!!.voteProgressBar.visibility = View.VISIBLE

            CoroutineScope(Dispatchers.IO).launch {
                val mutation =
                    GQLClient.instance.mutation(CreateVoteMutation(questionId, value)).execute()

                withContext(Dispatchers.Main) {
                    if (mutation.data?.votes?.create?.errors != null && mutation.data!!.votes.create.errors!!.isNotEmpty()) {
                        binding!!.submit.error = mutation.data!!.votes.create.errors!![0].message
                        binding!!.submit.visibility = View.VISIBLE
                        binding!!.voteProgressBar.visibility = View.GONE
                    } else {
                        val intent = Intent()
                        setResult(RESULT_OK, intent)
                        finish()
                    }

                }
            }

        }
    }
}